packages/wonder-blocks-tooltip/components/tooltip-arrow.js

```diff
@@ -0,0 +1,232 @@
+// @flow
+import * as React from "react";
+import {css, StyleSheet} from "aphrodite";
+
+import Colors from "@khanacademy/wonder-blocks-color";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+
+import type {Placement} from "../util/types.js";
+import type {PopperArrowProps} from "react-popper";
+
+type Props = {
+    placement: Placement,
+    popperArrowProps?: PopperArrowProps,
+};
+
+export default class TooltipArrow extends React.Component<Props> {
+    lastRef: ?HTMLElement;
+
+    updateRef(ref: ?HTMLElement) {
+        const {popperArrowProps} = this.props;
+        // We only want to update the popper's arrow reference if it is
+        // actually changed. Otherwise, we end up in an endless loop of updates
+        // as every render would trigger yet another render.
+        if (popperArrowProps && ref && ref !== this.lastRef) {
```

Comment:
How often are we re-rendering `TooltipArrow`?

Reply:
This is not to do with re-rendering from what I can tell. All I know is that without this, we are eternally called again because of calling the ref with the same value.

I suspect it may be down to how we're doing our anchoring without wrapping.

Reply:
So my suspicion is this.

1. We render and we call the ref to tell the popper what we are
2. The popper takes measurements and then calls to render us again with the changed positions

If we don't check that we are the same element, we re-render and RECALL the ref, which then signals the popper to redo its work.